From ca5eb69b780b0a7046234c6f0b505c6d445dbff3 Mon Sep 17 00:00:00 2001
From: Mender Team <mender@northern.tech>
Date: Fri, 17 Jul 2026 11:22:12 +0000
Subject: [PATCH] Integration of Mender boot code into U-Boot

Rebased onto u-boot-xlnx xlnx_rebase_v2025.01 (context offsets only, no
functional change) to pass the patch-fuzz QA check.
---
 include/env_default.h          |   3 +
 scripts/Makefile.autoconf      |   3 +-
 2 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/include/env_default.h b/include/env_default.h
index aa3dd40..ecb48f7 100644
--- a/include/env_default.h
+++ b/include/env_default.h
@@ -13,6 +13,8 @@
 
 #include <generated/environment.h>
 
+#include <env_mender.h>
+
 #ifdef DEFAULT_ENV_INSTANCE_EMBEDDED
 env_t embedded_environment __UBOOT_ENV_SECTION__(environment) = {
 	ENV_CRC,	/* CRC Sum */
@@ -27,6 +29,7 @@ char default_environment[] = {
 #else
 const char default_environment[] = {
 #endif
+	MENDER_ENV_SETTINGS
 #ifndef CONFIG_USE_DEFAULT_ENV_FILE
 #ifdef	CONFIG_ENV_CALLBACK_LIST_DEFAULT
 	ENV_CALLBACK_VAR "=" CONFIG_ENV_CALLBACK_LIST_DEFAULT "\0"
diff --git a/scripts/Makefile.autoconf b/scripts/Makefile.autoconf
index c1eab2f..092fb4b 100644
--- a/scripts/Makefile.autoconf
+++ b/scripts/Makefile.autoconf
@@ -116,7 +116,8 @@ define filechk_config_h
 	$(if $(CONFIG_SYS_CONFIG_NAME),echo \#include \<configs/$(CONFIG_SYS_CONFIG_NAME).h\> ;) \
 	echo \#include \<asm/config.h\>;				\
 	echo \#include \<linux/kconfig.h\>;				\
-	echo \#include \<config_fallbacks.h\>;)
+	echo \#include \<config_fallbacks.h\>;				\
+	echo \#include \<config_mender.h\>;)
 endef
 
 include/config.h: scripts/Makefile.autoconf create_symlink FORCE
